script/single-deployment/DeployAAVEBorrower.s.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.28;
+
+import {Script, console} from "forge-std/Script.sol";
+
+import {AaveBorrower} from "../../src/AaveBorrower.sol";
+
+import {DeployFlashLoanRouter, FlashLoanRouter} from "./DeployFlashLoanRouter.s.sol";
+
+import {Constants} from "script/libraries/Constants.sol";
+
+contract DeployAAVEBorrower is Script, DeployFlashLoanRouter {
+    function run() public virtual override(DeployFlashLoanRouter) {
+        deployAAVEBorrower(FlashLoanRouter(address(0)));
+    }
+
+    function deployAAVEBorrower(FlashLoanRouter router) internal returns (AaveBorrower borrower) {
+        address routerAddress = address(router) != address(0) ? address(router) : address(newFlashLoanRouter());
```

Comment:
I don't think this is necessary anymore. You can always simply compute the flashloan router address here (no need to pass it in) even in the deployAll case.

Also no need to convert into address and back (you keep the `FlashLoanRouter` type).